src/dawn/engine_dawn.cc

```diff
@@ -286,12 +291,48 @@ Result EngineDawn::DoClear(const ClearCommand*) {
   color_attachment.loadOp = ::dawn::LoadOp::Clear;
   color_attachment.storeOp = ::dawn::StoreOp::Store;
 
-  ::dawn::RenderPassDescriptor rpd =
-      device_->CreateRenderPassDescriptorBuilder()
-          .SetColorAttachments(1, &color_attachment)
-          .GetResult();
-  command_buffer_builder_.BeginRenderPass(rpd).EndPass();
-  return {};
+  ::dawn::RenderPassColorAttachmentDescriptor* rpca[] = {&color_attachment};
+  ::dawn::RenderPassDescriptor rpd;
+  rpd.colorAttachmentCount = 1;
+  rpd.colorAttachments = rpca;
+  rpd.depthStencilAttachment = nullptr;
+  ::dawn::RenderPassEncoder pass =
+      command_buffer_builder_.BeginRenderPass(&rpd);
+  pass.EndPass();
+
+  // Make sure we have a queue.
+  if (!queue_)
+    queue_ = device_->CreateQueue();
+
+  // Now run the commands.
+  ::dawn::Buffer& fb_buffer = render_pipeline_info_.fb_buffer;
+  auto command_buffer = command_buffer_builder_.Finish();
+
+  if (render_pipeline_info_.fb_data != nullptr) {
+    fb_buffer.Unmap();
+    render_pipeline_info_.fb_data = nullptr;
+  }
+
+  queue_.Submit(1, &command_buffer);
+
+  auto* pipeline = command->GetPipeline();
+  const std::vector<amber::Pipeline::BufferInfo>& out_color_attachment =
+      pipeline->GetColorAttachments();
+
+  // And any further commands start afresh.
+  DestroyCommandBufferBuilder();
+  MapResult map = MapBuffer(*device_, fb_buffer);
+
+  for (size_t i = 0; i < 1; ++i) {
+    auto& img = map;
+    auto& info = out_color_attachment[i];
+    auto* values = info.buffer->ValuePtr();
+    values->resize(info.buffer->GetSizeInBytes());
+    // assert map.size == info->buffer->GetSizeInBytes()
```

Comment:
instead of assert:

```
if (map.size != info->buffer->GetSizeInBytes())
  return Result("Dawn: framebuffer size does not match buffer size");
```

Reply:
Fixed with a TODO, since life is more complicated than this.